invenio_github/views/github.py

```diff
@@ -24,225 +24,221 @@
 
 from __future__ import absolute_import
 
-import json
-from datetime import datetime
-
-import humanize
-import pytz
-from dateutil.parser import parse
-from flask import Blueprint, abort, current_app, redirect, render_template, \
-    request, url_for
-from flask_babelex import gettext as _
+from functools import wraps
+
+from flask import Blueprint, abort, current_app, render_template
 from flask_breadcrumbs import register_breadcrumb
 from flask_login import current_user, login_required
 from flask_menu import register_menu
 from invenio_db import db
+from invenio_i18n import gettext as _
+from invenio_theme.proxies import current_theme_icons
+from speaklater import make_lazy_string
 from sqlalchemy.orm.exc import NoResultFound
 
-from ..api import GitHubAPI, GitHubRelease
-from ..errors import RepositoryAccessError
-from ..models import Release, Repository
-from ..proxies import current_github
-from ..utils import parse_timestamp, utcnow
+from invenio_github.api import GitHubAPI
+from invenio_github.proxies import current_github
 
-blueprint = Blueprint(
-    'invenio_github',
-    __name__,
-    static_folder='../static',
-    template_folder='../templates',
-    url_prefix='/account/settings/github',
-)
+from ..errors import GithubTokenNotFound, RepositoryAccessError
 
 
-#
-# Template filters
-#
-@blueprint.app_template_filter('naturaltime')
-def naturaltime(val):
-    """Get humanized version of time."""
-    val = val.replace(tzinfo=pytz.utc) \
-        if isinstance(val, datetime) else parse(val)
-    now = datetime.utcnow().replace(tzinfo=pytz.utc)
+def request_session_token():
+    """Requests an oauth session token to be configured for the user."""
 
-    return humanize.naturaltime(now - val)
+    def decorator(f):
+        @wraps(f)
+        def inner(*args, **kwargs):
+            github = GitHubAPI(user_id=current_user.id)
```

Comment:
question: this decorator instantiates a `GithubAPI` object, while the views that use the decorator start by instantiating it (again). Should we pass the instance? E.g.

```python
kwargs["github_api"] = github
return f(*args, **kwargs)
```